drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -428,7 +445,7 @@ function DragAndDropTemplates(configuration) {
                 (!ctx.last_action_correct) ? h("p", {}, gettext("Some of your answers were not correct.")) : null,
                 h("p", {}, gettext("Hints:")),
                 h("ul", {}, msgs.map(function(message) {
-                    return h("li", {innerHTML: message.message});
+                    return h("li", {innerHTML: gettext(message.message)});
```

Comment:
@afzaledx I can't see how `makemessages` is supposed to figure out the string to translate?

Reply:
@OmarIthawi, I am no longer working on this PR. @pomegranited might be able to help you on this.

Reply:
Thanks @afzaledx! Didn't realize that.

Reply:
No problem. As far as the extraction of strings marked for translation goes, I think it's a manual process so far.

Reply:
Thanks for replying here @afzaledx 😄  Hope you're doing well.

> I can't see how makemessages is supposed to figure out the string to translate?

@OmarIthawi  This XBlock is kind of a special case, because when you first create a DnDv2 Xblock, we have a demo that displays which uses a bunch of strings stored in [default_data.py](https://github.com/edx-solutions/xblock-drag-and-drop-v2/blob/313fc3b01dd9867687aa430e2b7e5e6da4627f84/drag_and_drop_v2/default_data.py), and that `message.message` string may be one of them.  We do this so that educators using the XBlock for the first time can have a translated experience, which helps them to use the XBlock for their own course.

When the educator then customizes it to suit their own needs for the course, they will likely change all of these strings, and they therefore will be outside the compiled translations library, and so will no longer translated.  We unfortunately can't automatically translate strings we don't know about in advance (and haven't come up with a way for educators to safely provide their own dynamically interpreted translation dictionaries), so that's the best we can do for now.

> As far as the extraction of strings marked for translation goes, I think it's a manual process so far.

Yes, that's correct.  There's a a PR in progress to automate this process using `make` targets: https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/172